packages/edit-navigation/src/components/name-display/index.js

```diff
@@ -1,23 +1,29 @@
 /**
  * WordPress dependencies
  */
+import { useContext } from '@wordpress/element';
 import { ToolbarGroup, ToolbarButton } from '@wordpress/components';
 import { BlockControls } from '@wordpress/block-editor';
 /**
  * Internal dependencies
  */
 import {
+	untitledMenu,
+	useMenuEntity,
 	useSelectedMenuData,
 	IsMenuNameControlFocusedContext,
 } from '../../hooks';
-import { useContext } from '@wordpress/element';
 
 import { sprintf, __ } from '@wordpress/i18n';
 export default function NameDisplay() {
-	const { menuName } = useSelectedMenuData();
+	const { menuId } = useSelectedMenuData();
+	const { editedMenu } = useMenuEntity( menuId );
 	const [ , setIsMenuNameEditFocused ] = useContext(
 		IsMenuNameControlFocusedContext
 	);
+
+	const menuName = editedMenu?.name ?? untitledMenu;
```

Comment:
Maybe we should check for "falsy" values here instead of "nullish". This way, it will display fallback when the menu name is empty.